### PR TITLE
Xmlwitch

### PR DESCRIPTION
**Pull request by mouadhkaabachi**

### CODE REVIEW

File: olapy/core/services/xmla_discover_tools.py [hunk 1]
```diff
@@ -52,94 +54,92 @@ def change_catalogue(self, new_catalogue):
...
+                        'xmlns:EX':
+                        'urn:schemas-microsoft-com:xml-analysis:exception',
+                        'xmlns:xsd': 'http://www.w3.org/2001/XMLSchema',
+                        'xmlns:xsi':
+                        'http://www.w3.org/2001/XMLSchema-instance'
+                    }):
+                with xml.row:
+                    xml.DataSourceName('sales')
+                    xml.DataSourceDescription('sales Sample Data')
+                    xml.URL('http://127.0.0.1:8000/xmla')
```
Comment (vindarel) [inline olapy/core/services/xmla_discover_tools.py]: beware harcoded urls and ports.

**Comment by vindarel** [comment]
Impressive and a pleasure to see xml strings going away.

I'm all for more explicit commit titles. Like "xmlwitch in xxx function".

and… is that well tested ??

File: olapy/core/services/xmla_discover_tools.py (hunk 1 above)
Comment (mouadhkaabachi) [inline olapy/core/services/xmla_discover_tools.py]: It's not dangerous, it's not going to change, anyway I will change that

**Comment by mouadhkaabachi** [comment]
all tests pass successfully, check circle

**Pull request merged**